asset-transfer-basic/application-gateway-java/src/main/java/application/java/App.java

```diff
@@ -0,0 +1,274 @@
+
+		System.out.println("*** Transaction committed successfully");
+	}
+
+	/**
+	 * Submit transaction asynchronously, allowing the application to process the
+	 * smart contract response (e.g. update a UI) while waiting for the commit
+	 * notification.
+	 */
+	private static void transferAssetAsync(Contract contract) throws GatewayException, CommitException {
+
+		System.out.println("\n" + "--> Async Submit Transaction: TransferAsset, updates existing asset owner");
+		SubmittedTransaction commit = contract.newProposal("TransferAsset").addArguments(assetId, "Saptha").build()
+				.endorse().submitAsync();
+
+		byte[] result = commit.getResult();
+		String oldOwner = new String(result, StandardCharsets.UTF_8);
+
+		System.out.println(
+				"*** Successfully submitted transaction to transfer ownership from " + oldOwner + " to Saptha");
+		System.out.println("*** Waiting for transaction commit");
+		Status status = commit.getStatus();
+		if (!status.isSuccessful()) {
+			throw new Error("Transaction " + status.getTransactionId() + " failed to commit with status code "
+					+ status.getCode());
+
+		}
+		System.out.println("*** Transaction committed successfully");
+
+	}
+
+	private static void readAssetById(Contract contract) throws GatewayException, CommitException {
+
+		System.out.println("\n" + "--> Evaluate Transaction: ReadAsset, function returns asset attributes");
+		byte[] evaluateResult = contract.evaluateTransaction("ReadAsset", assetId);
+		System.out.println("*** Result:" + prettyJson(evaluateResult));
+	}
+
+	/**
+	 * submitTransaction() will throw an error containing details of any error
+	 * responses from the smart contract.
+	 */
+	private static void updateNonExistentAsset(Contract contract) throws GatewayException, CommitException {
+
+		try {
+			System.out.println("\n"
+					+ "--> Submit Transaction: UpdateAsset asset70, asset70 does not exist and should return an error");
+			contract.submitTransaction("UpdateAsset", "asset70", "blue", "5", "Tomoko", "300");
+			System.out.println("******** FAILED to return an error");
+
+		} catch (EndorseException | SubmitException | CommitStatusException e) {
+			System.out.println("*** Successfully caught the error: " + "\n");
+			e.printStackTrace();
+			System.out.println("\n" + "Error Details: ");
+			for (ErrorDetail detail : e.getDetails()) {
+				System.out.println("\n" + "address: " + detail.getAddress() + "\n"
+						+ "mspId: " + detail.getMspId() + "\n" + "message:" + detail.getMessage() + "\n");
+			}
+			System.out.println("Transaction ID: " + e.getTransactionId());
+			
+		} catch (CommitException e) {
+			System.out.println("*** Successfully caught the error: " + e);
+			e.printStackTrace();
+			System.out.println("Transaction ID: " + e.getTransactionId() + " status code: " + e.getCode());
+		} catch (Throwable error) {
+			System.out.println("*** Successfully caught the error: " + error);
+			error.printStackTrace();
+		}
```

Comment:
Tricky one this... really we are only expecting an `EndorseException` because we've submitted a transaction with bad arguments so endorsement will fail, but it's good to demonstrate the different exceptions that can be raised. The only one I really have issue with is the `Throwable` because that isn't something we ever expect to get, so I don't think we should catch that at all.

Reply:
Thanks.. removed catch with Throwable